src/iso.c

```diff
@@ -0,0 +1,494 @@
+    memcpy(&(dr->interleave_gap_size), dr_dest + 27, sizeof(dr->interleave_gap_size));
+    memcpy((dr->vol_seq_num), dr_dest + 28, sizeof(dr->vol_seq_num));
+    memcpy(&(dr->len_identifier), dr_dest + 32, sizeof(dr->len_identifier));
+    bytes_read += 33;
+
+    memcpy((dr->file_identifier), dr_dest + 33, dr->len_identifier);
+    dr->file_identifier[dr->len_identifier] = '\0';
+    bytes_read += dr->len_identifier;
+
+    iso_media_seek(iso_p, bytes_read - 64, SEEK_CUR);
+    if (bytes_read % 2 == 1) { // Odd length, so we need to seek one bit to pad
+        iso_media_seek(iso_p, 1, SEEK_CUR);
+    }
+    return 1;
+}
+
+/**
+ * @brief Converts a character hex number into a base 10 integer
+ *
+ * @param src Array of hex characters in big endian format
+ * @param len Number of bytes in the src to convert
+ * @return The integer value of the first len bytes of hex characters in src
+ */
+static uint32_t hex_to_int(unsigned char *src, int len) {
+    uint32_t i, sum = 0;
+    for (i = 0; i < len; i++) {
+        sum *= 256;
+        sum += src[i];
+    }
+    return sum;
+}
+
+/**
+ * @brief Checks whether flags represent a directory
+ * @details Bit compares directory record file flags to the dir flag bit
+ *
+ * @param flags The file flags from a directory record
+ * @return 1 if dir bit is high, 0 otherwise
+ */
+static bool is_dir(int flags) {
+    if ((flags >> 1) & 1) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+static bool is_valid_record(struct directory_record *dr) {
+     if (dr->rec_date_time[3] < 0 || dr->rec_date_time[3] >= 24) {  // validate hours
+         return 0;
+     }
+     if (dr->rec_date_time[4] < 0 || dr->rec_date_time[4] >= 60) {  // validate minutes
+         return 0;
+     }
+     if (dr->rec_date_time[5] < 0 || dr->rec_date_time[5] >= 60) {  // validate seconds
+         return 0;
+     }
+     if (dr->rec_date_time[1] <= 0 || dr->rec_date_time[1] > 12) {  // validate month
+         return 0;
+     }
+     if (dr->rec_date_time[2] <= 0 || dr->rec_date_time[2] > 31) {  // validate day
+         return 0;
+     }
+     if (dr->rec_date_time[0] < 0) {    // validate year
+         return 0;
+     }
+     return 1;
+ }
+
+//Documentation in header
+int iso_dclose(struct iso_dir *dir) {
+    if (dir) {
+        kfree(dir);
+        return 0;
+    }
+    return -1;
+}
+
+//Documentation in header
+struct iso_dir *iso_dopen(const char *pname, int ata_unit) {
+    uint32_t dl;  //data length, needed in iso_look_up
+    int extent_num = iso_look_up(pname, &dl, ata_unit);
+    if (extent_num < 0) {
+        return 0;
+    }
+    struct iso_dir *to_return = kmalloc(sizeof(*to_return));
+    struct iso_point *iso_p = iso_media_open(ata_unit);
+    iso_media_seek(iso_p, ATAPI_BLOCKSIZE * extent_num, SEEK_SET);
+    struct directory_record *dr = kmalloc(sizeof(*dr));
+    get_directory_record(iso_p, dr);
+    if (!is_dir(dr->file_flags[0])) {
+        console_printf("Failed to open directory\n");
```

Comment:
Do we want to leave error messages in? This library will eventually be used through sys calls, and there won't be a console. Although stubbing out the console to a logging system once we hit user mode isn't a bad idea...